src/node_buffer.cc

```diff
@@ -1182,8 +1207,12 @@ void Swap16(const FunctionCallbackInfo<Value>& args) {
   THROW_AND_RETURN_UNLESS_BUFFER(env, args.This());
   SPREAD_ARG(args.This(), ts_obj);
 
-  for (size_t i = 0; i < ts_obj_length; i += 2) {
-    SWAP_BYTES(ts_obj_data, i, i + 1);
+  CHECK_EQ(ts_obj_length % 2, 0);
+  uint16_t* data16 = reinterpret_cast<uint16_t*>(ts_obj_data);
```

Comment:
@trevnorris sorry for the delay. I asked on some forums ([link](http://forums.codeguru.com/showthread.php?555959-Acceptable-use-of-undefined-behavior), [link](http://www.cplusplus.com/forum/general/192562/)) to try to make totally sure this is UB and that there's no way around it that doesn't involve memcpy/memmove/etc. Not hopeful :(.

- Again, the code works as-is, but it's your guys' call if you think it's acceptable to include this code given that it's UB. I don't know the full range of implications of doing so, but it doesn't sound like a great idea. If you guys are okay including it, I can add the tests and benchmarks for unaligned access. (For now: the perf is amazingly similar between aligned and unaligned.)
- I'm not super keen on adding something that will improve perf ~1/2 to 1/8th of the time, if I were to add a byte-wise fall-through to handle unaligned buffers. Thoughts?
- The alternative of writing inline assembly would work but is also unappealing for quite a few reasons, including portability and maintainability.
- I could just add the swap64 method using the existing `SWAP_BYTES` macro. My [node-bswap](https://github.com/zbjornson/node-bswap) module exists for cases requiring high perf, and is safe because typed arrays must be aligned.

Sorry for not catching this prior to submitting the PR!